core_brain/market_resolution.py

```
@@ -0,0 +1,597 @@
+    # dominant hold-to-resolution case has NO closes -- that is the one we
+    # settle. A condition that saw any close is skipped entirely rather than
+    # guessing how much of it remains unrealised.
+    try:
+        with registry._conn() as conn:
+            existing = conn.execute(
+                "SELECT 1 FROM closes WHERE lower(condition_id) = lower(?) LIMIT 1",
+                (condition_id,),
+            ).fetchone()
+        if existing:
+            return None
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)